code/lib/core-common/src/utils/resolve-path-in-sb-cache.ts

```diff
@@ -10,16 +10,8 @@ import pkgDir from 'pkg-dir';
  * @return {string} Absolute path to the file or directory
  */
 export function resolvePathInStorybookCache(fileOrDirectoryName: string): string {
-  const cwd = process.cwd();
-  const projectDir = pkgDir.sync(cwd);
-
-  let cacheDirectory;
-
-  if (!projectDir) {
-    cacheDirectory = path.resolve(cwd, '.cache/storybook');
-  } else {
-    cacheDirectory = path.resolve(projectDir, 'node_modules/.cache/storybook');
-  }
+  let cacheDirectory = findCacheDirectory({ name: 'storybook' });
+  cacheDirectory ||= path.join(process.cwd(), '.cache/storybook');
```

Comment:
The `findCacheDirectory` function's apparently can return `undefined` and this seemed like the most readable form of a fallback value while also keeping your code-style.